WF-CDPInstaller/plugin_dir/apache_plugin.py

```diff
@@ -0,0 +1,267 @@
+                        '  </Instance>\n').format(instance=sv_name,
+                                                  url=url_auto)
+                    utils.cprint()
+                    utils.cprint(
+                        'Your url is appended with ?auto to convert '
+                        'the content into machine readable code.\n'
+                        '{}'.format(plugin_instance))
+                    res = utils.ask(
+                        'Is this the correct status to monitor?')
+                    if res:
+                        utils.print_step('Saving instance')
+                        count = count + 1
+                        server_list.append(url)
+                        sv_list.append(sv_name)
+                        out.write(plugin_instance)
+                        utils.print_success()
+                    else:
+                        utils.cprint('Instance is not saved.')
+        out.write('</Plugin>\n')
+        return count
+
+    def check_apache_server_status(self, payload):
+        atitle_re = re.search(
+            r'(<head>[\w\W\s]*<title>Apache '
+            'Status</title>[\w\W\s]*</head>)([\w\W\s]*)', payload)
+
+        if atitle_re is not None:
+            res = atitle_re.group(2)
+            ah1_re = re.search(
+                r'<h1>(Apache Server Status for )([\w\W]*?)</h1>',
+                res)
+            if ah1_re is not None:
+                res = ah1_re.group(1) + ah1_re.group(2)
+                return res
+
+        return None
+
+    def include_apache_es_conf(self, conf_dir):
+        """
+        Assume extendedstatus.conf is unique to wavefront and writing over it.
+        """
+        filename = 'extendedstatus.conf'
+        filepath = '{conf_dir}/{filename}'.format(
+                  conf_dir=conf_dir, filename=filename)
+        content = (
+            '# ExtendedStatus controls whether Apache will generate '
+            '"full" status\n'
+            '# information (ExtendedStatus On) '
+            'or just basic information (ExtendedStatus\n'
+            '# Off) when the "server-status" handler is called. '
+            'The default is Off.\n'
+            'ExtendedStatus on')
+
+        out = utils.write_file(filepath, content)
+
+    def apache_plugin_usage(self):
+        utils.cprint(
+          'To monitor a apache server, '
+          'you must enable the server-status page.\n'
+          'To enable a server-status page, the following code in quote\n'
+          '"<Location /server-status>\n'
+          '  SetHandler server-status\n'
+          '</Location>"\n'
+          'must be included within a <VirtualHost> block '
+          'for the .conf file of your server.\n\n'
+          'To check whether the server-status page is working, '
+          'please visit\n'
+          '\tyour-server-name/server-status\n'
+          'It should look similar to\n'
+          '\tapache.org/server-status\n')
+if __name__ == '__main__':
```

Comment:
curious about these: what is the desired behavior if you just run these files as main?